### PR TITLE
Fix linking issue on Windows

### PR DESCRIPTION
**Pull request by gleocadie**
# What does this PR do?

This PR fixes a linking with for Windows native libraries that link against the VC runtime statically that was introduced by [the PR adding telemetry.
](https://github.com/DataDog/libdatadog/pull/339)
# Motivation

The .NET profiler links against the VC runtime statically so it can be deployed easily everywhere (on windows) without the customers having to install additional stuffs (VC runtime for example).

The PR that introduced the telemetry in the datadog_profiling lib, brought some new dependencies. Now the .NET profiler is unable to successfully build on windows.
The build produces this linking error:
```
datadog_profiling.lib(windows.o) : error LNK2019: unresolved external symbol __imp_strncpy referenced in function get_os_release [D:\a\libdatadog\libdatadog\examples\ffi\build_static\telemetry_metrics.vcxproj]
```

# Implementation

To fix this, I added a `.cargo/config.toml` file in the `profiling-ffi` crate and build the crate directly inside (not from the workspace, otherwise, we will have to move the `config.toml` in the workspace and it will apply to all crates)

# Additional Notes

Nop

# How to test the change?

I added in the CI 2 configurations when building and testing FFI:
- One where the projects are linked dynamically against the VC runtime
- One where the projects are linked statically against the VC runtime.

## For Reviewers
- [ ] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [ ] This PR doesn't touch any of that.

### CODE REVIEW

File: build-profiling-ffi.sh [hunk 1]
```diff
@@ -162,7 +162,6 @@ unexpected_native_libs=$(comm -13 <(echo "$unique_expected_libs") <(echo "$uniqu
 if [ -n "$unexpected_native_libs" ]; then
     echo "Error - More native static libraries are required for linking than expected:" 1>&2
     echo "$unexpected_native_libs" 1>&2
-    exit 1
```
[the remark this answers is not in the document]
Comment (gleocadie) [reply build-profiling-ffi.sh:165]: no you are right. It was during my debugging time (that's why it's still a draft :P)

**Review by r1viollet — APPROVED** [review]
LGTM

**Review by r1viollet — APPROVED** [review]
LGTM

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -115,8 +113,29 @@ jobs:
           path: ${{ github.workspace }}/artifacts
           retention-days: 1
 
+      - name: "(Windows) Test building C bindings - dynamic link vc runtime"
+        if: ${{ matrix.platform == 'windows-latest' }}
+        shell: bash
```
Comment (danielsn) [inline .github/workflows/test.yml:118]: It never ceases to amaze me that there is now supported bash on windows

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -115,8 +113,29 @@ jobs:
...
+        run: |
+          set -e
```
Comment (danielsn) [inline .github/workflows/test.yml:120]: for debugging, might be useful to `set -eux`

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -115,8 +113,29 @@ jobs:
...
+          cmake --build .
+
+      - name: "(Windows) Test building C bindings - static link vc runtime"
+        if: ${{ matrix.platform == 'windows-latest' }}
+        shell: bash
+        run: |
+          set -e
+          mkdir examples/ffi/build_static
+          cd examples/ffi/build_static
+          cmake -S .. -DDatadog_ROOT=$OUTPUT_FOLDER/profiling -DVCRUNTIME_LINK_TYPE=STATIC
```
Comment (danielsn) [inline .github/workflows/test.yml:133]: Do we also need to test FFI in both configurations?

File: build-profiling-ffi.sh [hunk 5]
```diff
@@ -100,15 +100,17 @@ sed < cmake/DatadogConfig.cmake.in \
...
 FEATURES="--features cbindgen,datadog-profiling-ffi/ddtelemetry-ffi"
 if [[ "$symbolizer" -eq 1 ]]; then
     FEATURES="--features cbindgen,datadog-profiling-ffi/ddtelemetry-ffi,symbolizer"
 fi
 
-DESTDIR="$destdir" cargo build --package="${datadog_profiling_ffi}" ${FEATURES} --release --target "${target}"
+# build inside the crate to use the config.toml file
+pushd profiling-ffi
+DESTDIR="$destdir" cargo build ${FEATURES} --release --target "${target}"
+popd
```
Comment (danielsn) [inline build-profiling-ffi.sh:113]: Would it be cleaner to 
```suggestion
( cd profiling-ffi && DESTDIR="$destdir" cargo build ${FEATURES} --release --target "${target}")
```

File: profiling-ffi/.cargo/config.toml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+[build]
+rustflags = ["-C", "relocation-model=pic"]
```
Comment (danielsn) [inline profiling-ffi/.cargo/config.toml:2]: Does this have any performance implications?

File: .github/workflows/test.yml (hunk 3 above)
Comment (danielsn) [reply .github/workflows/test.yml:120]: https://gist.github.com/mohanpedala/1e2ff5661761d3abd0385e8223e16425

File: profiling-ffi/.cargo/config.toml (hunk 6 above)
Comment (gleocadie) [reply profiling-ffi/.cargo/config.toml:2]: TBH, I do not know. I just moved this from the bash script

File: .github/workflows/test.yml (hunk 2 above)
Comment (gleocadie) [reply .github/workflows/test.yml:118]: Yeah Git bash is installed :) Not really conventional but it works 🤷

**Review by danielsn — APPROVED** [review]

**Pull request merged**